src/mpid/ch4/shm/shmam/eager/include/shmam_eager_impl.h

```diff
@@ -0,0 +1,110 @@
+/*
+ *  (C) 2006 by Argonne National Laboratory.
+ *      See COPYRIGHT in top-level directory.
+ *
+ *  Portions of this code were written by Intel Corporation.
+ *  Copyright (C) 2011-2016 Intel Corporation.  Intel provides this material
+ *  to Argonne National Laboratory subject to Software Grant and Corporate
+ *  Contributor License Agreement dated February 8, 2012.
+ */
+#ifndef SHM_SHMAM_EAGER_IMPL_H_INCLUDED
+#define SHM_SHMAM_EAGER_IMPL_H_INCLUDED
+
+#ifndef SHMAM_EAGER_DIRECT
+#ifndef SHMAM_EAGER_DISABLE_INLINES
+
+MPL_STATIC_INLINE_PREFIX int MPIDI_SHMAM_eager_init(int rank, int grank, int size)
+{
+    return MPIDI_SHMAM_eager_func->init(rank, grank, size);
```

Comment:
I'm not sure why we need these function pointers in the shmmod layer at all.  All possible shmmods are always compiled in (they have different namespaces).  This is different from the netmod architecture, where all netmods expose the same API.